### DISCUSSION TITLE
Ensure pipe tables

### DISCUSSION THREAD

**Discussion by halloleo** [ickc/pantable#32] [Q&A] [answered]
When using the `pantable` filter I try to generate *pipe tables* (because they are supported by other markdown converters like multimarkdown).

This doesn't work normally: All tables are generated as *grid tables*. However by accident I found out id *does* work when I set `width` to  a list of `0`s. Example:

    ``` {.table}
    ---
    alignment: DDD
    header: true
    markdown: true
    width: [0, 0, 0]
    ---
    First row,defaulted to be header row,can be disabled
    1,cell,It can be
    2,more,"It can be  very, very, very, very, very, very, very, very, very, very, very looooong cell."
    ```

Some questions about this: 

* Is this an indented feature? 
* Is it documented and i have overlooked it? 
* Is it just a lucky coincidence I should not rely on?

**Comment by halloleo** [comment]
Note: The command line I use is 

    pandoc -F pantable  input.md -t markdown-simple_tables+pipe_tables

**Comment by ickc** [comment, marked as answer]
It's happening on the pandoc side.

pantable parses the CSV into pandoc's native AST, and pandoc decided what to do about it for different outout, including markdown.

In pandoc. there's 4 table syntaxes, where only the grid table supports everything the internal AST support. So if it figures out a certain feature cannot be represented as pipe table, then it will uses grid table for example. And one such feature is the column width, I think.

Another way to force pandoc to output pipe table is to turn off the other three table extensions, such as `pandoc -t markdown-grid_table-...`. But this has a side effect: if pandoc decide a table cannot be rendered as pipe table, it will choose the next avaible format to output, in this case if other 3 extensions is turned off. the only remaining one will be HTML table (since HTML is valid markdown.)
